script.js

```diff
@@ -1,4 +1,696 @@
+var playerIndex = 0;
+var splitPlayer = {};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (mode == "playersetup" || mode == "inputname" || mode == "betmode") {
+    compHand = [];
+    return playerArraySetup(input);
+  }
+  if (mode == "dealmode" || mode == "compdeal") {
+    shuffledDeck = shuffleCards(deck);
+    var myOutputValue = dealCards(shuffledDeck);
+    return myOutputValue;
+  }
+  if (
+    mode == "playerturn" ||
+    mode == "playerDecisionTurn" ||
+    mode == "playerEndTurn" ||
+    mode == "splitDecision"
+  ) {
+    var playerTurnText = playerTurnFunc(input);
+    return playerTurnText;
+  }
+  if (mode == "compturn") {
+    var myOutputValue = compTurnFunc(compHand);
+  }
+  // need to return cards to the deck at the end of the game. Reshuffle happens in the dealcards function.
+  shuffledDeck = shuffleCards(deck);
   return myOutputValue;
 };
+
+var playerArraySetup = function (input) {
+  if (mode == "playersetup") {
+    if (Number.isNaN(Number(input)) || Number(input) <= 0) {
+      return `Sorry, please only input a number higher than 0.`;
+    }
+    dealIndex = 0;
+    compBlackjack = false;
+    cardDisplayCounter = 0;
+    noOfPlayers = Number(input);
+    allPlayers = [];
+    playerIndex = 0;
+    playerObject = {};
+    while (playerIndex < noOfPlayers) {
+      playerObject = {
+        player: {
+          username: "",
+          playerHand: [],
+          bankroll: 100,
+          currentBet: 0,
+          blackjackCheck: false,
+          handTotal: 0,
+          handText: "",
+          wasSplit: false,
+        },
+      };
+      allPlayers.push(playerObject);
+      playerIndex += 1;
+    }
+    mode = "inputname";
+    playerIndex = 0;
+    return `There are ${noOfPlayers} players. Please input your names in order.`;
+  }
+  if (mode == "inputname") {
+    allPlayers[playerIndex].player.username = input;
+    playerIndex += 1;
+    if (playerIndex < noOfPlayers) {
+      var myOutputValue = `Hi ${
+        allPlayers[playerIndex - 1].player.username
+      }, you are Player ${playerIndex}. Next player, please input your name.`;
+    } else {
+      mode = "betmode";
+      var myOutputValue = `Hi ${
+        allPlayers[playerIndex - 1].player.username
+      }, you are Player ${playerIndex}. Now, please be prepared to enter your bets in order starting from Player 1.`;
+      playerIndex = 0;
+    }
+    return myOutputValue;
+  } else if (mode == "betmode") {
+    if (allPlayers[playerIndex].player.bankroll <= 0) {
+      allPlayers.splice(playerIndex, 1);
+      if (playerIndex < noOfPlayers) {
+        noOfPlayers -= 1;
+        return `Sorry, you have gone bankrupt. You will not be able to play unless the game is restarted. Next player will now input their bet.`;
+      } else {
+        mode = "compdeal";
+        return `Sorry, you have gone bankrupt. You will not be able to play unless the game is restarted. The computer will deal the cards.`;
```

Comment:
your code here already has quite a few levels of nesting which makes it harder to understand. To make this easier to read, you can remove the else in line 103 and just write the code as per normal. This is because we are already returning from our if statement so only code that does not meet the condition of our if statement will continue to run. Hence, there is no need for an else here to explicitly state that only code that does not meet the condition will continue to run the following lines of code